BDArmory/FX/BulletHitFX.cs

```diff
@@ -28,14 +28,17 @@ public static ObjectPool CreateDecalPool(string modelPath)
             return ObjectPool.CreateObjectPool(template, BDArmorySettings.MAX_NUM_BULLET_DECALS, false, true, 0, true);
         }
 
-        public void AttachAt(Part hitPart, RaycastHit hit, Vector3 offset)
+        public void AttachAt(Part hitPart, RaycastHit hit, Vector3 offset, bool applyVelCorrection)
         {
             if (hitPart is null) return;
             parentPart = hitPart;
             // parentPartName = parentPart.name;
             // parentVesselName = parentPart.vessel.vesselName;
-            transform.SetParent(hitPart.transform);
-            transform.position = hit.point + offset;
+            // I assume the hit.collider should always exist, in any case, perform a null check for the
```

Comment:
I'm fairly certain that `hit.collider` should always exist for a valid hit (which is checked long before the code gets to here) and the `if (hitPart is null) return;` should catch any cases where the collider has disappeared due to the part disappearing.